pandas/core/internals/construction.py

```diff
@@ -242,7 +242,11 @@ def init_dict(data, index, columns, dtype=None):
 
         # no obvious "empty" int column
         if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
+            if (
+                dtype is None
+                or is_extension_array_dtype(dtype)
+                or np.issubdtype(dtype, np.flexible)
+            ):
```

Comment:
I _think_ changing this fixes the interval case

```suggestion
            if dtype is None or (
                not is_extension_array_dtype(dtype)
                and np.issubdtype(dtype, np.flexible)
            ):
```

Reply:
This suggestion doesn't work...

Reply:
i've pushed this change, works on my machine. can you elaborate?

Reply:
Thanks @simonjayhawkins . works on my environment too.